### ISSUE TITLE
Gomplate v3.11.x Go 1.21 panic: go4.org/unsafe/assume-no-moving-gc

### ISSUE THREAD

**Issue by erharb** [hairyhenderson/gomplate#1872]
Same as PR #1796 except that was only addressed in v4. Since v4 is not yet a stable release I think it should be ported back to the `3.11.x` branch as well.

```bash
go: downloading github.com/hairyhenderson/gomplate/v3 v3.11.5
...
panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.21 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH="go1.21" set. Notably, if go1.21 adds a moving garbage collector, this program is unsafe to use.
```

The solution should be to update the module to the version indicated in the PR. I may have to first seek approval to contribute from my org, so if someone beats me to it so be it!

```bash
go get go4.org/unsafe/assume-no-moving-gc@e7c30c7
go mod tidy
```

**Comment by hairyhenderson** [comment]
@erharb are you building `gomplate` yourself? If so, why not build `v4`? Or why not use a pre-built binary?

**Comment by erharb** [comment]
Company security controls mostly, we have our own goproxy for scanning and handling go mods/installs. Usually it's just more paperwork to bring in pre-compiled binaries built from the internet, rather than building it in-house where possible for our internal usage.

I wanted to start with a "stable" released version first since we're evaluating using gomplate for use with GitHub Template repositories, though I'll definitely give v4 a shot soon!

**Comment by evanfreed** [comment]
Confirmed using v4 similar to [here](https://github.com/hairyhenderson/gomplate/pull/1796#issuecomment-1681455853) is a sufficient workaround.

**Comment by giorgiga** [comment]
@hairyhenderson I maintain the gomplate package for nixos and this same issue currently breaks the it. I'd love to upgrade to v4 but the latest released version is v3.11.5... will v4 be released soon? (I'd rather not go to the hoops necessary to build from git if possible, mainly because I'd have to investigate what those hoops are exactly).

**Comment by hairyhenderson** [comment]
> Company security controls mostly [...]
😬 sorry to hear that @erharb - v3.11.5 should be more "stable", though I think v4 is _stable enough_ (just not complete at this point).

> I maintain the gomplate package for nixos and this same issue currently breaks it.
Thanks for that detail @giorgiga (and thanks for maintaining that package!).

I have some vacation time coming up next week, so I'll try to get a "clean" 3.11.6 released then.

v4 is still a ways out - I just haven't had the free time to dedicate to it for a while 😞

**Comment by giorgiga** [comment]
Thanks @hairyhenderson - for the moment I've updated the package to force using go 1.20.

Unfortunately I'm not much of a go programmer, so I can't be of much help on the actual issue :(

**Comment by hairyhenderson** [comment]
Fixed by #1910 - I'll release 3.11.6 soon...

**Comment by giorgiga** [comment]
That's great - thanks @hairyhenderson !!
